src/pages/n/[id].tsx

```diff
@@ -64,22 +65,30 @@ export const EditorPage: NextPage = () => {
     (newScrollState: ScrollState) => {
       if (scrollSource.current === ScrollSource.RENDERER && editorSyncScroll) {
         setScrollState((old) => {
-          const newState = { editorScrollState: newScrollState, rendererScrollState: old.rendererScrollState }
-          log.debug('Set scroll state because of renderer scroll', newState)
-          return newState
+          const newState: DualScrollState = {
+            editorScrollState: newScrollState,
+            rendererScrollState: old.rendererScrollState
+          }
+          return equal(newState, old) ? old : newState
```

Comment:
If newState and old are equal why do you return the old state?
Always returning newState should be okay.

Reply:
If I return new state then the returned object would be content equal but it's not same object instance.
By returning old state I suppress a rendering because of an state update.